pkg/extensions/config/sync/config.go

```diff
@@ -33,7 +33,8 @@ type RegistryConfig struct {
 	RetryDelay       *time.Duration
 	OnlySigned       *bool
 	CredentialHelper string
-	PreserveDigest   bool // sync without converting
+	PreserveDigest   bool          // sync without converting
+	SyncTimeout      time.Duration // timeout for on-demand sync operations; if zero or unset, defaults to 3 hours
```

Comment:
If we want to set a default, I suggest configuring it in pkg/cli/server/root.go, function applyDefaultValues, together with the other defaults.